### DISCUSSION TITLE
8bitdo SN30 Pro support

### DISCUSSION THREAD

**Discussion by jim150** [darthcloud/BlueRetro#34] [Q&A] [answered]
First of all thank you for this amazing project.
I’ve build the GC adapter and worked perfectly with the Switch Pro and PS4 controllers.
On the other hand, I’ve tested it with the 8bitdo SN30 Pro (in Switch mode) and it didn’t fully work for me, the buttons worked but the stick did not.
Also I’ve tried a Wii U clone controller and it didn’t pair at all.

**Comment by darthcloud** [comment, marked as answer]
Use it in xinput mode (Start + X) IIRC.

**Comment by jim150** [comment]
Yes, in this mode worked perfectly! Thanks

**Reply by darthcloud** [reply]
Thanks a lot for the  coffee 😀!!!

**Reply by Cyrus331** [reply]
“8bitdo SN30 Pro ”
Does it support vibration? I can use it normally on PS2, but there is no vibration. It is also set on the IO setting page. Is there anything else to set?
